src/TLuaInterpreter.cpp

```diff
@@ -10739,6 +10750,520 @@ int TLuaInterpreter::restartIrc(lua_State* L)
+ *  Returns the queued text at index.
+ *  If no index exists, returns false.
+ *  If no index provided, returns a table with the entire queue.
+ */
+int TLuaInterpreter::ttsGetSpeechQueue(lua_State* L)
+{
+    TLuaInterpreter::ttsBuild();
+
+    if (lua_gettop(L) > 0) {
+        if (!lua_isnumber(L, 1)) {
+            lua_pushfstring(L, "ttsGetspeechQueue. bad argument #1 type (optional index as number expected, got %s!)", luaL_typename(L, 1));
+            lua_error(L);
+            return 1;
+        }
+
+        int index = lua_tonumber(L, 1);
+        index--;
+
+        if (index < 0 || index > speechQueue.size()) {
+            lua_pushboolean(L, false);
+            return 1;
+        }
+
+        lua_pushstring(L, speechQueue.at(index).toLatin1().constData());
+        return 1;
+    }
+
+    lua_newtable(L);
+
+    for (int i = 0; i < speechQueue.size(); i++) {
+        lua_pushnumber(L, i + 1);
+        lua_pushstring(L, speechQueue.at(i).toUtf8().constData());
+        lua_settable(L, -3);
+    }
+
+    return 1;
+}
+
+/** ttsPauseSpeech()
+ *  Pauses speech.
+ */
+int TLuaInterpreter::ttsPauseSpeech(lua_State* L)
+{
+    TLuaInterpreter::ttsBuild();
+
+    speechUnit->pause();
+
+    return 0;
+}
+
+/** ttsResumeSpeech()
+ *  Resumes speech.
+ */
+int TLuaInterpreter::ttsResumeSpeech(lua_State* L)
+{
+    TLuaInterpreter::ttsBuild();
+
+    speechUnit->resume();
+
+    return 0;
+}
+
+/** ttsClearQueue( opt. index )
+ *  Removes an item from the speech queue at index.
+ *  Returns false if invalid index provided.
+ *  Clears entire queue if no index provided.
+ */
+int TLuaInterpreter::ttsClearQueue(lua_State* L)
+{
+    TLuaInterpreter::ttsBuild();
+
+    if (lua_gettop(L) > 0) {
+        if (!lua_isnumber(L, 1)) {
+            lua_pushfstring(L, "ttsClearQueue: bad argument #1 type (optional index as int expected, got %s!)", luaL_typename(L, 1));
+            lua_error(L);
+            return 1;
+        }
+
+        int index = lua_tonumber(L, 1);
+        index--;
+
+        if (index < 0 || index >= speechQueue.size()) {
+            lua_pushfstring(L, "ttsClearQueue: index (%d) out of bounds for queue size %d", index + 1, speechQueue.size());
+            lua_error(L);
+            return 1;
+        }
+
+        speechQueue.remove(index);
+        return 0;
+    }
+
+    speechQueue.clear();
+    return 0;
+}
+
+/** ttsGetCurrentLine()
+ *  Returns the text being spoken, or false if not speaking.
+ */
+int TLuaInterpreter::ttsGetCurrentLine(lua_State* L)
+{
+    TLuaInterpreter::ttsBuild();
+
+    if (speechUnit->state() == QTextToSpeech::Ready || speechUnit->state() == QTextToSpeech::BackendError) {
+        lua_pushboolean(L, false);
+        return 1;
+    }
+
+    lua_pushstring(L, speechCurrent.toUtf8().constData());
+    return 1;
+}
+
+/** ttsGetState()
+ *  Returns the current state of the speechUnit.
+ *  Possible returns: ttsSpeechReady, ttsSpeechPaused, ttsSpeechStarted, ttsSpeechError, ttsUnknownState.
```

Comment:
I'd like to have these as enum values similar to the buttons for the gamepad if possible.